### PR TITLE
+effis installation version

### PR DESCRIPTION
**Pull request by suchyta1**
I added a `+effis` variant to install (non-passthrough) XGC and GENE with EFFIS turned on, and made some improvements to the EFFIS package file. A `spack install effis` is still needed for the composition front-end, but that of course can't be added to the same package due to the single concretization.

### CODE REVIEW

File: spack/wdmapp/packages/wdmapp/package.py [hunk 1]
```diff
@@ -56,3 +52,21 @@ class Wdmapp(BundlePackage):
...
+    # CWS - If the variant is not specified for hdf5 then there are
+    # concretization errors associated with hdf5. I think this is related 
+    # to https://github.com/spack/spack/issues/15478 .
+
+    # Contretization fixes
+    depends_on('adios  +fortran +sz')
+    depends_on('adios2 +fortran +sz')   # The ADIOS +sz (+fortran) is to be explicit, but it's the default
+    depends_on('sz@:1.4.12.99')         # ADIOS1 needs sz < 2.0.0, ADIOS2 defaults to sz > 2.0.0
+    depends_on('hdf5 +hl +fortran')     # PETSc needes +hl, Apps need +fortran
+    depends_on('python@:2.9.99')        # PETSc 3.7.7 needs Python 2.7, EFFIS defaults to Python 3
```
Comment (germasch) [inline spack/wdmapp/packages/wdmapp/package.py:71]: Did you try whether this works with spack v14.2? @cwsmith and I had troubles where we had to list these identical dependencies for each variant -- but maybe by listing them last, not first, spack works as intended? If so, that'd be good.

Comment (suchyta1) [reply spack/wdmapp/packages/wdmapp/package.py:71]: I didn't try v0.14.2. There's definitely sometimes still oddities in what happens if specs aren't explicitly given everywhere. I wouldn't be surprised if you need to repeat for each variant. If you notice, I did that in the EFFIS package. Though, I think that was needed becasue `python-type=full` is being set from `packages.yaml`, and I don't think that fully settles out concretization issues in all cases.

**Comment by suchyta1** [comment]
> This looks good to me (but the one question).
> 
> I'm wondering, though, whether we really need effis to be a variant, rather than just always requiring effis. I suppose always enabling effis shouldn't have too much of a downside -- it doesn't really introduce more dependencies, so build-wise, it shouldn't be an issue to have it on.
> 
> I'm less sure about whether the pragmas change behavior when running without effis, though? I suppose that by default it'll still read adios2cfg.cml and behave the same?
> 
> The main reason is that we don't really have any good automated testing in the first place, so we more or less rely on people building/running things by hand, and the more variants there are, the greater the chance that untested variants will stop working at some point....

If there's no `+effis` variant, the pragmas are never dealt with, which is needed, say, for the plotting. (Currently using `+effis` requires then doing the run with EFFIS composition. But I should be able to fix that in the future.)

**Comment by germasch** [comment]
> If there's no `+effis` variant, the pragmas are never dealt with, which is needed, say, for the plotting. (Currently using `+effis` requires then doing the run with EFFIS composition. But I should be able to fix that in the future.)

Right, I understood what `+effis` is needed for -- I just thought we could always enable it for GENE and XGC, not conditionally on `+effis` in wdmapp. But since you're saying that compiling XGC/XGC with +effis means they can't run without effis anymore, I guess we do need the variant. I think you should default it to on, though.

I'm testing whether this still works with spack v14.2 right now, looking good so far.

**Comment by germasch** [comment]
Looks like both`+effis` and `~effis` builds worked on Summit, so good to go other than making `+effis` default.

**Comment by suchyta1** [comment]
> Looks like both`+effis` and `~effis` builds worked on Summit, so good to go other than making `+effis` default.

Thanks! I tested both on Rhea too.

**Pull request merged**